modules/era/lua_dynamis/globals/era_dynamis.lua

```diff
@@ -1344,8 +1344,10 @@ end
 
 m:addOverride("xi.dynamis.qmOnTrigger", function(player, npc) -- Override standard qmOnTrigger()
     local zoneId = npc:getZoneID()
-    player:addKeyItem(xi.dynamis.dynaInfoEra[zoneId].winKI)
-    player:messageSpecial(zones[zoneId].text.KEYITEM_OBTAINED, xi.dynamis.dynaInfoEra[zoneId].winKI)
+    if not player:hasKeyItem(xi.dynamis.dynaInfoEra[zoneId].winKI) then
+        player:addKeyItem(xi.dynamis.dynaInfoEra[zoneId].winKI)
+        player:messageSpecial(zones[zoneId].text.KEYITEM_OBTAINED, xi.dynamis.dynaInfoEra[zoneId].winKI)
```

Comment:
We can use npcUtil to give a key item. This gives the key item and gives the message. Simple simplification of code into one line.

Reply:
> We can use npcUtil to give a key item. This gives the key item and gives the message. Simple simplification of code into one line.

Thanks, made the simplification!